WordPress/gutenberg
# Implemented HoC ifPropsVerify and IfPostTypeSupports.

ifPropsVerify HoC was implemented, it will be useful in a considerable amount of places. IfPostTypeSupports HoC is a specialization of ifPropsVerify The IfPostTypeSupports replaces PostTypeSupportCheck component, using an HoC makes things easier to test as we don't need to declare more than one component per file as happened in PageAttributesOrder.
A replacement of IfPostTypeSupports was added here as a sample. Provided the new HoC's are approved and merged, all PostTypeSupportCheck and (similar components) will be replaced with HoC's.

Documentation by @aduth:
**Example Before:**

```jsx
function MyFeaturedImage() {
	return (
		<PostTypeSupportCheck supportKeys="thumbnail">
			<img />
		</PostTypeSupportCheck>
	);
}

export default connect( /* ... */ )( MyFeaturedImage );
```

**Example After:**

```jsx
function MyFeaturedImage() {
	return <img />;
}

export default compose( [
	ifPostTypeSupports( 'thumbnail' ),
	connect( /* ... */ ),
] )( MyFeaturedImage );
```

__Implementation notes:__

The thinking behind the naming convention is that we could create higher-order component-libraries with the following expectations:

Name|Behavior
---|---
`ifSomeCondition`|Renders component only if condition applies
`withSomeProps`|Renders component with additional props

__Testing instructions:__

Verify that there are no regressions in the behavior of post-type-conditional elements:

- Discussion (`comments` or `trackbacks` supports)
- Featured image (`thumbnail` support)
- Excerpt (`excerpt` support)
- Post format (`post-formats` support)
- Author (`author` support)
- Order (`page-attributes` support)
- Last Revision (`revisions` support)
